llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -2767,7 +2766,9 @@ SDValue AMDGPUTargetLowering::LowerFLOGCommon(SDValue Op,
   EVT VT = Op.getValueType();
   SDNodeFlags Flags = Op->getFlags();
   SDLoc DL(Op);
-
+  // Our implementation of LOG is not contract safe, so disable instruction
+  // contraction.
+  Flags.setAllowContract(false);
```

Comment:
Can you sink this down into the actual implementation, for the sequences where it's a problem